src/store/slices/docs.ts

```diff
@@ -61,5 +82,30 @@ export const docsSlice: StoreSlice<IAppStateDocsSlice> = (set, get) => ({
     return selected.includes(doc);
   },
 
-  clearSelected: () => set((state) => ({ docs: { ...state.docs, selected: [] } })),
+  // fully clear the selected list
+  clearSelected: () =>
+    set((state) => ({ docs: { ...state.docs, selected: [], isAllSelected: false } }), false, 'docs/clearSelected'),
+
+  // make the new selection the union between the two lists
+  selectAll: () =>
+    set(
+      (state) => ({
+        docs: { ...state.docs, selected: union(state.docs.selected, state.docs.current), isAllSelected: true },
+      }),
+      false,
+      'docs/selectAll',
+    ),
+
+  // remove items that exist in both lists, but keep the others in `selected` intact
+  clearAllSelected: () =>
+    set(
+      (state) => ({
+        docs: { ...state.docs, selected: difference(state.docs.selected, state.docs.current), isAllSelected: false },
```

Comment:
I don't really understand the purpose of the function here. Is it suppose to clear all so nothing is selected? So `selected` should be empty?

Reply:
So I should have thought up a better name/way to do it.

But the problem is that we want an overall selection that persists as you page through results, but have the `Select All` functionality work page to page.  So it should clear after each pagination.  So we want to only clear the docs that exist in both lists, but leave the rest of `selected` intact.
So in a way, there shouldn't be a time when we're completely clearing the selected docs (unless we are doing a new search, for example)